test/python/utest_masarclient/test_masarClient-mongo.py

```diff
@@ -0,0 +1,204 @@
+import sys
+import unittest
+
+from masarclient import masarClient
+from masarclient.channelRPC import epicsExit
+from pymasarmongo.db import utils
+from pymasarmongo.config._config import masarconfig
+
+from pymasarmongo.pymasarmongo.pymasar import saveconfig
+from pymasarmongo.pymasarmongo.pymasar import retrieveconfig
+from pymasarmongo.pymasarmongo.pymasar import updateconfig
+from pymasarmongo.pymasarmongo.pymasar import saveevent
+from pymasarmongo.pymasarmongo.pymasar import retrieveevents
+from pymasarmongo.pymasarmongo.pymasar import updateevent
+from pymasarmongo.pymasarmongo.pymasar import retrievesnapshot
+
+'''
+
+Unittests for masarService/python/masarclient/masarClient.py
+
+'''
+
+
+class TestMasarClientMongo(unittest.TestCase):
+
+    def setUp(self):
+        channel = 'mongoMasarTestService'
+        self.mc = masarClient.client(channelname=channel)
+        # DB SETUP
+        self.conn, collection = utils.conn(host=masarconfig.get('mongodb', 'host'), port=masarconfig.get('mongodb', 'port'),
+                                      db=masarconfig.get('mongodb', 'database'))
+
+        self.conn.drop_database(masarconfig.get('mongodb', 'database'))
+        name = "SR_All_20140421"
+        params = {"desc": "SR daily SCR setpoint without IS kick and septum: SR and RF",
+                  "system": "SR",
+                  "status": "active",
+                  "version": 20140421,
+                  }
+        newid = saveconfig(self.conn, collection, name, **params)
+        #res0 = retrieveconfig(self.conn, collection, name)
+        pvs = ["masarExample0000",
+               "masarExample0001",
+               #"masarExampleBoUninit",
+               "masarExampleMbboUninit",
+               "masarExample0002",
+               "masarExample0003",
+               "masarExample0004",
+               "masarExampleCharArray",
+               "masarExampleShortArray",
+               "masarExampleLongArray",
+               "masarExampleStringArray",
+               "masarExampleFloatArray",
+               "masarExampleDoubleArray",
+               "masarExampleMbboUninitTest"]
+        # TODO: Save will fail if list contains only 1 PV
+        updateconfig(self.conn, collection, name, pvlist={"names": pvs})
+        #res3 = retrieveconfig(self.conn, collection, name, withpvs=True)
+
+    def tearDown(self):
+        self.conn.drop_database(masarconfig.get('mongodb', 'database'))
+        utils.close(self.conn)
+
+    def testRetrieveSystemList(self):
+        result = self.mc.retrieveSystemList()
+        self.assertNotEqual(result, None)  # Can not be assertTrue because there is no case where it returns True
+        self.assertNotEqual(result, False)  # Instead asserting both not None and not False
```

Comment:
As you've discovered, equality can be a tricky concept.  For this situation I'd recommend a more positive test.  Perhaps ```self.assertIsInstance(result, (list, tuple))```.  Failing that ```self.assertNotIn(result, (None, False))``` would be more concise.